### PR TITLE
TM-5944-Agenda-Item-Delete-API

### PR DESCRIPTION
**Pull request by cagbakwu**
Dual Merge: 
- [FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2969)

[Ticket](https://metaphase.atlassian.net/browse/TM-5944)

### CODE REVIEW

File: talentmap_api/fsbid/urls/agenda.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 urlpatterns = [
     url(r'^agenda_items/(?P<pk>[0-9]+)/$', views.AgendaItemView.as_view(), name="agenda-agenda-item"),
+    url(r'^agenda_items/delete/$', views.AgendaItemDeleteView.as_view(), name="agenda-delete-agenda-item"),
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/urls/agenda.py:10]:
```suggestion
    url(r'^agenda_item/delete/$', views.AgendaItemDeleteView.as_view(), name="agenda-delete-agenda-item"),
```

File: talentmap_api/fsbid/views/agenda.py [hunk 2]
```diff
@@ -22,8 +23,25 @@ def get(self, request, pk):
...
-
-
+    
+class AgendaItemDeleteView(BaseView):
+    permission_classes = [Or(isDjangoGroupMember('cdo'), isDjangoGroupMember('ao_user'),)]
+    
+    @swagger_auto_schema(
+        manual_parameters=[
+            openapi.Parameter("aiseqnum", openapi.IN_QUERY, type=openapi.TYPE_INTEGER, description='Panel Meeting ID'),
+            openapi.Parameter("aiupdatedate", openapi.IN_QUERY, type=openapi.TYPE_STRING, description='Panel Meeting Date'),
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/views/agenda.py:33]: update descriptions

File: talentmap_api/fsbid/views/agenda.py [hunk 3]
```diff
@@ -22,8 +23,19 @@
...
-
+    def post(self, request):
+        '''
+        Delete single agenda by ai_seq_num
+        '''
+        try:
+            services.delete_agenda_item(request.data, request.META['HTTP_JWT'])
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        except Exception as e:
+            return Response(str(e), status=status.HTTP_422_UNPROCESSABLE_ENTITY)
```
[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/views/agenda.py:44]: @cagbakwu

File: talentmap_api/fsbid/services/agenda.py [hunk 4]
```diff
@@ -309,6 +309,14 @@ def edit_agenda_item(query, jwt_token):
         **args
     )
 
+def delete_agenda_item(query, jwt_token):
+    '''
+    Deletes agenda item
+    '''
+    ai_seq_num = query.get("aiseqnum")
+    ai_update_date = query.get("aiupdatedate")
+    url = f"{API_ROOT}/v1/agendas/{ai_seq_num}/?aiupdatedate={ai_update_date}"
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/agenda.py:318]:
```suggestion
    url = f"{API_ROOT}/v1/agendas/{ai_seq_num}?aiupdatedate={ai_update_date}"
```

File: talentmap_api/fsbid/views/agenda.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cagbakwu) [reply talentmap_api/fsbid/views/agenda.py]: @elizabeth-jimenez oops!, changed it!

**Review by elizabeth-jimenez — APPROVED** [review]

**Pull request merged**